app/value_objects/conviction_type.rb

```diff
@@ -38,7 +38,7 @@ def self.find_constant(raw_value)
     PROHIBITION                        = new(:prohibition,                      parent: COMMUNITY_ORDER, calculator_class: Calculators::YouthRehabilitationOrderCalculator),
     REFERRAL_ORDER                     = new(:referral_order,                   parent: COMMUNITY_ORDER, calculator_class: Calculators::YouthRehabilitationOrderCalculator),
     REHAB_ACTIVITY_REQUIREMENT         = new(:rehab_activity_requirement,       parent: COMMUNITY_ORDER, calculator_class: Calculators::YouthRehabilitationOrderCalculator),
-    REPARATION_ORDER                   = new(:reparation_order,                 parent: COMMUNITY_ORDER),
+    REPARATION_ORDER                   = new(:reparation_order,                 parent: COMMUNITY_ORDER, calculator_class: Calculators::ImmediatelyCalculator),
```

Comment:
This `reparation_order` conviction subtype should have the `skip_length: true` because is spent immediately (as the `absolute_discharge`). So we don't have to ask for the length. I think this was an oversight.

Then we can update the spec so the double doesn't need `conviction_length` or `conviction_length_type` methods. And maybe also cukes?